docs/blueprints/EP035.rst

```diff
@@ -0,0 +1,281 @@
+
+KytosEvent ``topology.interruptions.start``
+===========================================
+
+Notify services of an interruption starting,
+after the interruption provider has already updated the status.
+
+Expected Input:
+
+.. code-block:: python3
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+Example Usage:
+
+.. code-block:: python3
+
+    # Creating events when updating status
+
+    def start_switch_maintenance(switches: set[str]):
+        affected_switches = switches - maintenance_switches
+        maintenance_switches += switches
+        controller.buffers.app.put(
+            KytosEvent(
+                name='topology.interruptions.start'
+                content={
+                    'type': 'maintenance',
+                    'switches': affected_switches,
+                }
+            )
+        )
+
+
+
+KytosEvent ``topology.interruptions.end``
+=========================================
+
+Notify services of an interruption ending,
+after the interruption provider has already updated the status.
+
+.. code-block:: python3
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+Example Usage:
+
+.. code-block:: python3
+
+    # Creating events when updating status
+
+    def end_switch_maintenance(switches: set[str]):
+        affected_switches = switches & maintenance_switches
+        maintenance_switches -= switches
+        controller.buffers.app.put(
+            KytosEvent(
+                name='topology.interruptions.end'
+                content={
+                    'type': 'maintenance',
+                    'switches': affected_switches,
+    }
+            )
+        )
+
+Notes:
+
+- Upon the end of an interruption, services which had there best available paths interrupted should make an effort to reclaim interrupted paths.
+
+
+New Topology REST Endpoints
+###########################
+
+The following set of new endpoints are intended to allow
+for previewing the affects of adding in interruptions.
+
+REST Endpoint: ``GET <url for previewing>``
+======================================
+
+For the given service provider,
+previews services affected if the specified interruption occurs now.
+
+Expected Input:
+
+.. code-block:: json
+
+    {
+        "type": <interruption type str>,
+        "switches": [<switch id>],
+        "interfaces": [<interface id>],
+        "links": [<link id>],
+    }
+
+Expected Output:
+
+.. code-block:: json
+
+    [
+        {
+            'id': <id>,
+            'name': <service_name | null>,
+            'owner': <str | null>,
+        },
+    ]
+
+REST Endpoint ``GET /topology/v3/preview_interruption``
+=======================================================
+
+For all known service providers,
```

Comment:
In the future we could add which `<service provider name>` to preview for, and dispatch async with `httpx`, currently we only have `mef_eline` so it's worth implementing the filter since in practice it won't make much of a different, but let's make sure that this uses `httpx.AsyncClient` to play well with coroutines if we add another srevice provider in the future, OK?